Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs

```diff
@@ -828,13 +828,28 @@ private void ApplySuppression(Pawn pawn)
             {
                 suppressionAmount = def.projectile.GetDamageAmount(1);
                 var propsCE = def.projectile as ProjectilePropertiesCE;
-                var penetrationAmount = propsCE?.armorPenetrationSharp ?? 0f;
-                var armorMod = penetrationAmount <= 0 ? 0 : 1 - Mathf.Clamp(pawn.GetStatValue(CE_StatDefOf.AverageSharpArmor) * 0.5f / penetrationAmount, 0, 1);
+                var damageArmorCategory = def.projectile.damageDef.armorCategory;
+                if ((propsCE.damageDef.GetModExtension<DamageDefExtensionCE>() ?? new DamageDefExtensionCE()).isAmbientDamage) //Ambient damage projectiles won't deal suppression (because I'm too lazy to implement average heat and electric armor, besides, being on fire already makes pawns panic), so let's skip this entirely.
+                {
+                    return;
+                }
+                //Good luck trying to untangle this mess.
+                float penetrationAmount = ((propsCE != null) ?
+                    (
+                        ((damageArmorCategory == CE_DamageArmorCategoryDefOf.Blunt) ?
+                            ((!propsCE.damageFalloff || propsCE.explosionRadius == 0) ?
+                                Mathf.Max(def.projectile.GetDamageAmount(1) * 0.3f, propsCE.armorPenetrationBlunt) :
+                                Mathf.Lerp(Mathf.Max(def.projectile.GetDamageAmount(1) * 0.3f, propsCE.armorPenetrationBlunt), 0.6f, Mathf.Pow(pawn.Position.DistanceTo(Position) / propsCE.explosionRadius, 0.55f)) //This is basically a condensed version of ExplosionCE.GetArmorPenetrationAtCE. Any changes to GetArmorPenetrationAtCE should be also applied to this.
+                            ) :
+                            propsCE.armorPenetrationSharp
+                        )
+                    ) : 0f
+                );
```

Comment:
I would suggest to  untangling this XD. we don't want future generations to hate us more.